environments/contrail/node-specific-hieradata.yaml

```diff
@@ -0,0 +1,73 @@
+# "contrail_analytics_database_node_ips": ["Contrail-v3.2-Analytics-DB-IP"],
+# "contrail_database_node_ips": ["Contrail-v3.2-Controller-IP"]}, 
+# "Contrail-v3.2-Analytics-DB-UUID": {"contrail::contrail_version": 3, 
+# "contrail_analytics_database_node_ips": ["Contrail-v3.2-Analytics-DB-IP"],
+# "contrail_database_node_ips": ["Contrail-v3.2-Controller-IP"],
+# "contrail_analytics_database_short_node_names": ["Contrail-v3.2-Analytics-DB-SHORT-HOSTNAME"]},
+# "Compute-Node-UUID": {"contrail::contrail_version": 3, "contrail_config_node_ips": ["Contrail-v3.x-Controller-IP"]}, 
+# "Contrail-v4.y-Controller-UUID": {"contrail_database_node_ips": ["Contrail-v4.y-Controller-IP"], 
+# "contrail::rabbit_vhost": "<rmq-vhost-for-new-cluster>", "contrail_config_node_ips": ["Contrail-v4.y-Controller-IP"],
+# "contrail_analytics_node_ips": ["Contrail-v4.y-Analytics-IP"]}, 
+# "Contrail-v4.y-Analytics-UUID": {"contrail_analytics_database_node_ips": ["Contrail-v4.y-Analytics-DB-IP"], 
+# "contrail_database_node_ips": ["Contrail-v4.y-Controller-IP"], "contrail::rabbit_vhost": "<rmq-vhost-for-new-cluster>",
+# "contrail_config_node_ips": ["Contrail-v4.y-Controller-IP"], "contrail_analytics_node_ips":
+# ["Contrail-v4.y-Analytics-IP"]}, 
+# "Contrail-v4.y-Analytics-DB-UUID": {"contrail_analytics_database_node_ips": ["Contrail-v4.y-Analytics-DB-IP"], 
+# "contrail_database_node_ips": ["Contrail-v4.y-Controller-IP"],
+# "contrail_analytics_database_short_node_names": ["Contrail-v4.y-Analytics-DB-SHORT-HOSTNAME"],
+# "contrail::rabbit_vhost": "<rmq-vhost-for-new-cluster>", "contrail_analytics_node_ips":
+# ["Contrail-v4.y-Analytics-IP"]}}
+# 
+# The sample UUID's/IP's/Names provided below are from a working sample. Keep the resource_registry section as
+# it is, and make changes to the NodeDataLookup section according to your overcloud deployment.
+#
+# The UUID-to-node mapping used in the following example is:
+# 003FEF0B-FC16-4C2E-B7EA-1B30008948A8: Contrail-3.2 Contrail-Analytics-Database node
+# 99FD12A9-191C-41B3-BF6B-3CE30C63CC01: OpenStack Controller node
+# D4B82082-5D64-4674-93AB-7377B6311820: Contrail-3.2 Contrail Controller node
+# 8350F888-1BB8-4CE9-9F48-096CAD3F39A9: Contrail-3.2 Contrail Analytics node
+# AB15DEA1-89B8-4621-9F79-5A6AE25BCF93: Compute node
+# AC0ED5FE-4B6E-4ED5-8FB7-A960E53F03B1: Contrail-4.0 Contrail Controller node
+# 38354BE7-77BD-4A8A-8C7C-3BED64A05684: Contrail-4.0 Contrail Analytics node
+# ED773F6E-AA03-41FD-A6CF-649205071D70: Contrail-4.0 Contrail Analytics-Database node
+#
+resource_registry:
+  OS::TripleO::ContrailAnalyticsDatabaseExtraConfigPre: ../../puppet/extraconfig/pre_deploy/per_node.yaml
+  OS::TripleO::ContrailAnalyticsExtraConfigPre: ../../puppet/extraconfig/pre_deploy/per_node.yaml
+  OS::TripleO::ContrailControllerExtraConfigPre: ../../puppet/extraconfig/pre_deploy/per_node.yaml
+  OS::TripleO::ComputeExtraConfigPre: ../../puppet/extraconfig/pre_deploy/per_node.yaml
+
```

Comment:
It looks it is needed to add extra config for ContrailDpdk and ContrailTsn as well.